libs/RobotKit/include/RobotController.h

```diff
@@ -27,11 +29,12 @@ class RobotController : public interface::RobotController
 
 	void raise(auto event) { state_machine.process_event(event); };
 
+	void initializeComponents() { _ble.init(); }
```

Comment:
est-ce qu'on peut pas le faire dans le constructeur? plutôt que d'appeler une fonction qu'on peut oublier d'appeler ? ou alors le ctor appelle la fonction privée `initializeComponents`.

je me pose la même question pour `registerEvents`

Reply:
On a déjà eu dans le passé des cas où le constructeur qui devait appeler des méthodes privées ne le faisait pas à l'exécution. On peut prendre le pari d'essayer.

Reply:
> On a déjà eu dans le passé des cas où le constructeur qui devait appeler des méthodes privées ne le faisait pas à l'exécution

on se souvient desquels et de pourquoi ça marchait pas?

ça serait intéressant de comprendre pourquoi, s'il fallait modifier un truc de notre côté ou si c'est "comme ça".

si pas le choix, on vivra avec. mais si c'est possible, ça rend l'interface et l'utilisation beaucoup plus simple et propre.

Reply:
Je vais essayer de reproduire le phénomène

On avait pris le parti d'être explicite et d'appeler les méthodes qui fallait plutôt qu'implicite comme pourrait l'être un nom de méthode générique `init();`

Il peut y avoir une réponse du côté de `most vexing parse` [[Source 1](https://stackoverflow.com/questions/180172/default-constructor-with-empty-brackets)] [[Source 2](https://stackoverflow.com/questions/11963849/c-constructor-not-being-called)]

Reply:
Après, c'est aussi pour les tests: c'est plus simple de focaliser sur ce qu'on attend de chaque fonctions au lieu d'avoir tous les appels et de devoir faire un `EXPECT_CALL` pour les unexpected calls

On peut les garder sous forme de fonctions, mais ça peut prêter à confusion d'avoir des fonctions publique qu'on appelle pas dans l'exécution du programme et uniquement en test

Reply:
Aussi, il semblerait que les tests soient aussi cassés

Même si les `EXPECT_CALL` sont définis, ils ne sont pas bien pris en compte à l'exécution des tests et on se retrouve avec 2 types d'erreurs: d'un côté il y a des uninterresting calls et de l'autre des EXPECTED_CALL qui ne sont pas appelés.

Je vais voir s'il y a quelque chose à faire, dans ce cas je supprimerai ce commentaire